lib/MySQL_PreparedStatement.cpp

```
@@ -50,14 +51,33 @@ static uint64_t stmt_compute_hash(char *user,
 	l += strlen(_COMPUTE_HASH_DEL2_);
 
 	// write query
+	char *nquery = NULL;
+	if (strip_query_comments) {
+		char *nquery = mysql_query_strip_comments((char *)query, query_length, false);
+		if (nquery != NULL) {
+			query = nquery;
+			query_length = strlen(nquery);
+		}
+	}
 	memcpy(buf + l, query, query_length);
```

Cppcheck (2.19.0)

[error] If memory allocation fails

(nullPointerArithmeticOutOfMemory)